### DISCUSSION TITLE
Multiselect Combobox issue

### DISCUSSION THREAD

**Discussion by LechK** [twilio-labs/paste#2994] [Ideas] [answered]
Hello Paste Team!
I found another issue in Multiselect Combobox component. (later MC)
In MC component we have props like `onInputValueChange` and `onSelectedItemsChange`.
With first one we can filter list of options, with second - we select to selections list.
One of requirements is, we write to inputValue to filter list and find item we are looking for, on selectedChange, we add that item to selection and we should drop inputValue. But that what is happening in MC (screenshots added):
1. We are typing what we are looking for (1 screenshot)
2. Click on item we would like to add to selection. (2 ssh)
3. inputValue is cleared and filter is dropped. (2ssh)
4. but text in MC is still present. (2ssh)
Could it be problem with controlled/uncontrolled component values? Or I'm doing something wrong?

`onInputValueChange={({ inputValue: newInputValue = "" }) => {
        setInputValue(newInputValue);
      }}`
`onSelectedItemsChange={(selectedItems) => {
        setInputValue("");
        setItems((prev) => ({ ...prev, items: selectedItems }));
      }}`
`inputValue={inputValue}`
      
      this is my code. Any ideas? Or it should be fixed in MC component?
![test1](https://user-images.githubusercontent.com/46899047/215456629-eeb2a7ae-0962-47a1-b5ee-e4b12e043376.png)
![Test2](https://user-images.githubusercontent.com/46899047/215456635-47099114-64bc-4b0b-ae07-e239c9ff6a27.png)

**Comment by gloriliale** [comment]
Hi @LechK! Thanks for bringing this up, it looks like a bug in the component- we've made a ticket to address it and will update here when it's all set!

**Comment by SiTaggart** [comment, marked as answer]
We've just released a new version of multiselect comboobox that allows you full access to the internals of the component. 

Multi select combobox can now be fully controlled by the consumer via inversion of control and using the provided hooks. Documentation can be found here https://paste.twilio.design/components/multiselect-combobox#usemultiselectcombobox-state-hook

Hopefully this will allow you to address this
